rscripts/YAL5-single-codons.R

```diff
@@ -0,0 +1,1108 @@
+                                           filter_for_frame = filter_for_frame,
+                                           snapdisp = snapdisp
+  )
+
+  return(total_codon_pos_counts)
+}
+
+
+# total_codon_pos_counts <- suppressMessages(GetAllCodonPosCounts(gene_names,
+# dataset, hd_file, min_read_length, filtering_frame, filter_for_frame))
+
+#TEST: GetAllCodonPosCounts(): returns a tibble.
+#TEST: GetAllCodonPosCounts(): the tibble has 3 columns.
+#TEST: GetAllCodonPosCounts(): the column names are %in% c("Gene",
+#                                          "PosCodon" and "Count").
+#TEST: GetAllCodonPosCounts(): number of observations in the output tibble
+#              = sum of CDS (codon co-ordinates) for all genes in gene_names.
+#TEST: GetAllCodonPosCounts(): the unique gene names in column "Gene" match the
+# genes in gene_names (unique(total_codon_pos_counts$Gene) = gene_names) = TRUE.
+#gives:
+# > str(total_codon_pos_counts)
+# Classes "tbl_df", "tbl" and "data.frame":   2749 observations of 3 variables:
+#   $ Gene    : chr  "YAL003W" "YAL003W" "YAL003W" "YAL003W" ...
+#   $ PosCodon: int  1 2 3 4 5 6 7 8 9 10 ...
+#   $ Count   : num  4249 825 1017 1176 1116 ...
+#
+#
+# #TEST:: GetAllCodonPosCounts example with tinysim, filter_for_frame = FALSE
+#
+#
+# Gene      PosCodon  Count
+# 1 MAT       1         0
+# 2 MAT       2         2
+# 3 MAT       3         2
+# 4 MAT       4         0
+# 5 MIKE      1         0
+# 6 MIKE      2         1
+# 7 MIKE      3         0
+# 8 MIKE      4         0
+# 9 MIKE      5         0
+#
+## TEST:: GetAllCodonPosCounts example with tinysim, filter_for_frame = TRUE
+#
+# A tibble: 9 x 3
+# Gene  PosCodon Count
+# <chr>    <int> <dbl>
+#   1 MAT          1     0
+# 2 MAT          2     0
+# 3 MAT          3     2
+# 4 MAT          4     0
+# 5 MIKE         1     0
+# 6 MIKE         2     0
+# 7 MIKE         3     0
+# 8 MIKE         4     0
+# 9 MIKE         5     0
+#
+# this is expected as only reads mapping to the first nucleotide of the codon
+# are retained
+
+
+### Functions to add codon identities to positions ###
+
+#' AddCodonNamesToCodonPosCounts(): Adds codon identities to position and counts
+#'
+#' @param gene_names The list of all the genes in the sample 
+#' @param gene The gene from gene_names to get read lengths for.
+#' @param dataset The name of dataset stored in .h5 file.
+#' @param hd_file The path to the .h5 hdf5 file holding read data for all genes,
+#' created from BAM files for dataset samples.
+#' @param gff_df The gff file in a dataframe format
+#' @param min_read_length An integer, the minimum read length in H5 output;
+#' Default = 10 (set in generate_stats_figs.R from yaml)
+#' @param filter_for_frame Decides which method of asite assignment to use
+#' @param filtering_frame The frame from which to select reads from.
+#' @param snapdisp An integer for additional displacement in the snapping
+#'
```

Comment:
tsv file is listed as an input for the function but is not present in the param list